File: 448/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int N = nums.size();
        int appear[N+1];
        for(int i=1; i<=N; i++) appear[i] = 0;
        for(int i=0; i<N; i++){
            int n = nums[i];
            appear[n]++;
        }

        vector<int> ans;
        for(int i=1; i<=N; i++){
            if(appear[i]==0) ans.push_back(i);
        }
        return ans;
    }
};

File: 448/find-all-numbers-disappeared-in-an-array.py
# LeetCode 448. Find All Numbers Disappeared in an Array
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        ans = []
        numSet = set(nums)
        N = len(nums)
        for i in range(1,N+1):
            if i not in numSet:
                ans.append(i)
        return ans
